src/sql/src/session/vars.rs

```diff
@@ -1938,6 +1942,35 @@ impl SessionVars {
         local: bool,
     ) -> Result<(), VarError> {
         let name = UncasedStr::new(name);
+        self.check_read_only(name)?;
+
+        self.vars
+            .get_mut(name)
+            .map(|v| {
+                v.visible(&self.user, system_vars)?;
+                v.set(input, local)
+            })
+            .transpose()?
+            .ok_or_else(|| VarError::UnknownParameter(name.to_string()))
+    }
+
+    /// Sets the "role default" parameter named `name` to the value represented by `value`.
+    ///
+    /// A role default only takes effect at the start of a Session, so running
+    /// `ALTER ROLE ... SET ...` has no visible impact until a new connection is started.
+    pub fn set_role_default(&mut self, name: &str, input: VarInput) -> Result<(), VarError> {
+        let name = UncasedStr::new(name);
+        self.check_read_only(name)?;
+
+        self.vars
+            .get_mut(name)
+            .map(|v| v.set_role_default(input))
```

Comment:
Do we need to check if the variable is visible? Can you add a test with an internal variable.